airflow/executors/kubernetes_executor.py

```diff
@@ -668,6 +669,22 @@ def _adopt_completed_pods(self, kube_client: kubernetes.client.CoreV1Api):
             except ApiException as e:
                 self.log.info("Failed to adopt pod %s. Reason: %s", pod.metadata.name, e)
 
+    def _handle_zombied_istio_pods(self, kube_client: client.CoreV1Api) -> None:
+        """
+        Handle Zombied pods that are caused because istio container is still running,
+        while base container (where Airflow task is run) is completed.
+
+        :param kube_client: kubernetes client for speaking to kube API
+        """
+        kwargs = {
+            'field_selector': "status.phase=Running",
+            'label_selector': 'kubernetes_executor=True',
```

Comment:
We could and probably should filter this further to only checks pod where the label/annotation for Scheduler_job_id != Current job id

Reply:
I think we'd rather want to look at any scheduler_job_id that isn't still running given HA. Unless I'm mistaken we'd need to loop to do that, and over time it's probably faster to simply iterate all running pods instead.

This does raise the question of a possible race between this adoption logic handling the sidecar and the normal executor though. Makes me wonder if both sides need to be in a try:except (though I think it is more likely to happen on this side). Also with HA, more than 1 could try handling these zombies at nearly the same time too.

Reply:
Agreed